token-handler/src/lib.rs

```diff
@@ -23,4 +23,25 @@ pub trait TokenHandler:
 
     #[upgrade]
     fn upgrade(&self) {}
+
+    #[only_owner]
+    #[endpoint(setEnshrineEsdtWhitelist)]
+    fn set_enshrine_esdt_whitelist(
```

Comment:
This should be `whitelist_enshrine_esdt` and pass only one contract address.
Also add a TODO to make this endpoint callable only by factorySC or configSC, I'm not sure which one.